InfiniteBlanK3T(2).CodingTracker/MainCodingTracker/DatabaseCreation.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.Data.Sqlite;
+using System.Configuration;
+
+namespace CodingTracker
+{    
+    public class DatabaseCreation
+    {
+        Validation val = new();
+        private string _name;
+        string connectionString = ConfigurationManager.ConnectionStrings["connectionstring"].ConnectionString;
+        
+        public DatabaseCreation(string name) 
+        {
+            _name = name;                                             
+            CheckMainTableExist();            
+        }
+
+        public DatabaseCreation() : this("Thomas_Default") { }
+
+        public string Name
+        {
+            set { _name = value; }
+            get { return _name; }
+        }
+
+        public void CheckMainTableExist()
+        {
+            using var connection = new SqliteConnection(connectionString);
+            
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText =
+                    @$"CREATE TABLE IF NOT EXISTS {Name} (
+                    Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                    Date TEXT,
+                    StartTime TEXT,
+                    EndTime TEXT,
+                    Duration INTEGER
+                    );";
+            try
+            {
+                tableCmd.ExecuteNonQuery();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Oh no! An error occured.\n - Details: " + ex.Message);
+            }
+        }
+
+        public void CheckGoalTableExist()
+        {
+            using var connection = new SqliteConnection(connectionString);
+
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText =
+                $@"CREATE TABLE IF NOT EXISTS Goals (
+                Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                Name TEXT,
+                Date TEXT,
+                TimePerDay Interger,
```

Comment:
Did you end up implementing the 'goal' feature? All good if you didn't since it's not a requirement, but there's a typo in 'interger'